hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -65,19 +65,21 @@ case class TableRead(path: String, spec: TableSpec, typ: TableType, dropRows: Bo
 
   def execute(hc: HailContext): TableValue = {
     val globals = spec.globalsComponent.readLocal(hc, path, typ.globalType)(0)
-    TableValue(typ,
+    val rvd = if (dropRows)
+      OrderedRVD.empty(hc.sc, typ.rvdType)
+    else {
+      val rvd = spec.rowsComponent.read(hc, path, typ.rowType)
+      (rvd, typ.key) match {
+        case (ordered: OrderedRVD, Some(k))
+          if ordered.typ.key.startsWith(k) => ordered
+        case (_, Some(k)) => rvd.toOrderedRVD.changeKey(k)
```

Comment:
trying to understand what this is doing --

Since the left is `_`, the rvd is an UnpartitionedRVD, so `toOrderedRVD` is a no-op cast, right? Then we coerce?

Reply:
we should also warn here -- this will help users know when they need to rewrite things.

Reply:
Either the rvd is an UnpartitionedRVD, or it's an OrderedRVD with a key that is not compatible with `k`. The `toOrderedRVD` is always a no-op, it just converts between `UnpartitionedRVD` and the corresponding `OrderedRVD` with empty key. The point is just to let met write all the actual logic as it will be once UnpartitionedRVD is gone. Actually, I shouldn't be doing the match on the type of `rvd`, I'll clean this up.